src/test/java/com/github/checkstyle/regression/git/DiffParserTest.java

```diff
@@ -20,47 +20,112 @@
+    public void testIsProperUtilsClass() throws Exception {
+        assertUtilsClassHasPrivateConstructor(DiffParser.class);
+    }
 
-    @Before
-    public void setUp() throws Exception {
-        repository = GitUtils.createNewRepository();
-        final File helloWorld = GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorld");
-        GitUtils.createNewBranchAndCheckout(repository, "foo");
-        Files.write(helloWorld.toPath(), "hello world!".getBytes(), StandardOpenOption.APPEND);
-        GitUtils.addAllAndCommit(repository, "append text to HelloWorld");
+    @Test
+    public void testParseAddChange() throws Exception {
+        try (Repository repository = GitUtils.createNewRepository()) {
+            GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorld");
+            GitUtils.createNewBranchAndCheckout(repository, "foo");
+            GitUtils.addAnEmptyFileAndCommit(repository, "AddedFile");
+            final List<GitChange> changes = DiffParser.parse(
+                    repository.getDirectory().getParent(), "foo");
+            assertEquals(1, changes.size());
+            assertChange(ADD, "AddedFile", changes.get(0));
+            FileUtils.deleteDirectory(repository.getDirectory().getParentFile());
+        }
     }
 
-    @After
-    public void tearDown() throws Exception {
-        repository.close();
+    @Test
+    public void testParseModifyChange() throws Exception {
+        try (Repository repository = GitUtils.createNewRepository()) {
+            final File helloWorld = GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorld");
+            GitUtils.createNewBranchAndCheckout(repository, "foo");
+            Files.write(helloWorld.toPath(), "hello world!".getBytes(), StandardOpenOption.APPEND);
+            GitUtils.addAllAndCommit(repository, "append text to HelloWorld");
+            final List<GitChange> changes = DiffParser.parse(
+                    repository.getDirectory().getParent(), "foo");
+            assertEquals(1, changes.size());
+            assertChange(MODIFY, "HelloWorld", changes.get(0));
+            FileUtils.deleteDirectory(repository.getDirectory().getParentFile());
+        }
     }
 
     @Test
-    public void testIsProperUtilsClass() throws Exception {
-        assertUtilsClassHasPrivateConstructor(DiffParser.class);
+    public void testParseDeleteChange() throws Exception {
+        try (Repository repository = GitUtils.createNewRepository()) {
+            GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorld");
+            GitUtils.addAnEmptyFileAndCommit(repository, "FileToDelete");
+            GitUtils.createNewBranchAndCheckout(repository, "foo");
+            GitUtils.removeFileAndCommit(repository, "FileToDelete");
+            final List<GitChange> changes = DiffParser.parse(
+                    repository.getDirectory().getParent(), "foo");
+            assertEquals(1, changes.size());
+            assertChange(DELETE, "/dev/null", changes.get(0));
```

Comment:
Where did `/dev/null` come from?
The file that was deleted was `FileToDelete` so I expect that to be the file name.
This file name makes no sense if you are planning to parse/use the file name for anything.